### PR TITLE
Remove AppConstants usage from Basilisk

### PR DESCRIPTION
**Pull request by mattatobin**
Tag: #300

### CODE REVIEW

**Comment by janekptacijarabaci** [comment]
IMHO/AFAIK:

https://github.com/MoonchildProductions/moebius/pull/310/files#diff-4ebd62319cbfe7ca966979473af5e2c7L66
https://github.com/MoonchildProductions/moebius/pull/310/files#diff-9b1dfdf129f1bca1ec1eecad0c9ab9a7L3316
https://github.com/MoonchildProductions/moebius/pull/310/files#diff-7d497354f9d223b84c9b205a51d956dcL206
https://github.com/MoonchildProductions/moebius/pull/310/files#diff-ce02bb53d129ac1bdec10a9ea87bd7daL838
bad:
\#ifdef
good:
\#ifndef

https://github.com/MoonchildProductions/moebius/pull/310/files#diff-22dbb24a7257410179904e34f5cd62d3R43
I suggest: Add an indent (2 spaces).

**Comment by mattatobin** [comment]
Thank you very much @janekptacijarabaci for catching those!

I am going to check through line by line with SxS to see if I missed anything else.

**Comment by janekptacijarabaci** [comment]
You are welcome.

File: application/basilisk/modules/RecentWindow.jsm [hunk 1]
```diff
@@ -36,25 +35,23 @@ this.RecentWindow = {
...
+    // if we're lucky, this isn't a popup, and we can just return this
+    if (win && !isSuitableBrowserWindow(win)) {
+      win = null;
+    let windowList = Services.wm.getEnumerator("navigator:browser");
+    // this is oldest to newest, so this gets a bit ugly
+    while (windowList.hasMoreElements()) {
+      let nextWin = windowList.getNext();
+      if (isSuitableBrowserWindow(nextWin))
+        win = nextWin;
     }
```
[the remark this answers is not in the document]
Comment (mattatobin) [reply application/basilisk/modules/RecentWindow.jsm]: k

**Comment by mattatobin** [comment]
At this point, I have gone through every line and it looks right to me and nits seem to be resolved.. I am just waiting on @wolfbeast to sign off on it.

File: application/basilisk/components/migration/EdgeProfileMigrator.js [hunk 2]
```diff
@@ -409,7 +408,7 @@ EdgeProfileMigrator.prototype.getLastUsedDate = function() {
  * See MigrationUtils.jsm for slightly more info on how sourceProfiles is used.
  */
 EdgeProfileMigrator.prototype.__defineGetter__("sourceProfiles", function() {
-  let isWin10OrHigher = AppConstants.isPlatformAndVersionAtLeast("win", "10");
+  let isWin10OrHigher = Services.vc.compare(Services.sysinfo.getProperty("version"), "10") >= 0;
```
[the remark this answers is not in the document]
Comment (mattatobin) [reply application/basilisk/components/migration/EdgeProfileMigrator.js]: Redundant, application/basilisk/components/migration/moz.build says this file is only included for Windows targets.

File: application/basilisk/components/migration/IEProfileMigrator.js [hunk 3]
```diff
@@ -111,7 +110,7 @@ IE7FormPasswords.prototype = {
 
   get exists() {
     // work only on windows until 7
-    if (AppConstants.isPlatformAndVersionAtLeast("win", "6.2")) {
+    if (Services.vc.compare(Services.sysinfo.getProperty("version"), "6.2") >= 0) {
```
[the remark this answers is not in the document]
Comment (mattatobin) [reply application/basilisk/components/migration/IEProfileMigrator.js]: Redundant, application/basilisk/components/migration/moz.build says this file is only included for Windows targets.

File: application/basilisk/components/migration/IEProfileMigrator.js [hunk 4]
```diff
@@ -348,7 +347,7 @@ IEProfileMigrator.prototype.getResources = function IE_getResources() {
     MSMigrationUtils.getCookiesMigrator(),
   ];
   // Only support the form password migrator for Windows XP to 7.
-  if (AppConstants.isPlatformAndVersionAtMost("win", "6.1")) {
+  if (Services.vc.compare(Services.sysinfo.getProperty("version"), "6.1") >= 0) {
```
[the remark this answers is not in the document]
Comment (mattatobin) [reply application/basilisk/components/migration/IEProfileMigrator.js]: Redundant, application/basilisk/components/migration/moz.build says this file is only included for Windows targets.

File: application/basilisk/components/migration/MSMigrationUtils.jsm [hunk 5]
```diff
@@ -723,7 +722,7 @@ WindowsVaultFormPasswords.prototype = {
 
   get exists() {
     // work only on windows 8+
-    if (AppConstants.isPlatformAndVersionAtLeast("win", "6.2")) {
+    if (Services.vc.compare(Services.sysinfo.getProperty("version"), "6.2") >= 0) {
```
[the remark this answers is not in the document]
Comment (mattatobin) [reply application/basilisk/components/migration/MSMigrationUtils.jsm]: Redundant, application/basilisk/components/migration/moz.build says this file is only included for Windows targets.

File: application/basilisk/components/migration/MigrationUtils.jsm [hunk 6]
```diff
@@ -745,10 +745,11 @@ this.MigrationUtils = Object.freeze({
       Cu.reportError("Could not detect default browser: " + ex);
     }
 
+#ifdef XP_WIN
     // "firefox" is the least useful entry here, and might just be because we've set
     // ourselves as the default (on Windows 7 and below). In that case, check if we
     // have a registry key that tells us where to go:
-    if (key == "firefox" && AppConstants.isPlatformAndVersionAtMost("win", "6.2")) {
+    if (key == "firefox" && Services.vc.compare(Services.sysinfo.getProperty("version"), "6.2") >= 0) {
```
[the remark this answers is not in the document]
Comment (mattatobin) [reply application/basilisk/components/migration/MigrationUtils.jsm]: Redundant, the entire block is `#ifdef XP_WIN`

File: application/basilisk/components/nsBrowserGlue.js [hunk 7]
```diff
@@ -946,7 +946,9 @@ BrowserGlue.prototype = {
     if (!disableResetPrompt && lastUse &&
         Date.now() - lastUse >= OFFER_PROFILE_RESET_INTERVAL_MS) {
       this._resetProfileNotification("unused");
-    } else if (AppConstants.platform == "win" && !disableResetPrompt) {
```
[the remark this answers is not in the document]
Comment (mattatobin) [reply application/basilisk/components/nsBrowserGlue.js]: Fixed.

File: application/basilisk/base/content/browser.js [hunk 8]
```diff
@@ -6343,7 +6354,13 @@ function warnAboutClosingWindow() {
   // OS X doesn't quit the application when the last window is closed, but keeps
```
[the remark this answers is not in the document]
Comment (mattatobin) [reply application/basilisk/base/content/browser.js]: Fixed

File: application/basilisk/components/shell/ShellService.jsm [hunk 9]
```diff
@@ -26,20 +27,17 @@ let ShellServiceInternal = {
    * environments.
    */
   get canSetDesktopBackground() {
-    if (AppConstants.platform == "win" ||
```
[the remark this answers is not in the document]
Comment (mattatobin) [reply application/basilisk/components/shell/ShellService.jsm]: Fixed to be effectively the same as the original control structure but without creating an unreachable code after return

**Comment by wolfbeast** [comment]
Looks good now. merging...

**Comment by adeshkp** [comment]
Mac build is busted. Now, it failes with:
`64:51.24 mozbuild.preprocessor.Error: ('/Users/admin/labs/hard/moebius/application/basilisk/base/content/browser-feeds.js', 341, 'UNDEFINED_VAR', 'MOZ_MACBUNDLE_NAME')`


Few notable warnings (possibly unrelated):
`64:29.95 /Users/admin/labs/hard/moebius/jetpack/source/python-lib/cuddlefish/xpi.py:22: UserWarning: Duplicate name: 'locale/'`
`64:37.27 /Users/admin/labs/hard/moebius/application/basilisk/components/migration/SafariProfileMigrator.js: WARNING: no preprocessor directives found`

**Comment by mattatobin** [comment]
The moz.build file in that component needs a local define set for MOZ_MACBUNDLE_NAME

**Comment by adeshkp** [comment]
Mac build fixed in #320

**Pull request merged**